pkg/snapshot/overlay.go

```diff
@@ -347,7 +347,7 @@ func (o *snapshotter) Prepare(ctx context.Context, key, parent string, opts ...s
 
 	// if parent is not empty, try to attach and mount block device
 	_, writableBD := info.Labels[LabelSupportWritableOverlayBD]
-	if parent != "" {
+	if _, ok := info.Labels[labelKeyTargetSnapshotRef]; !ok && parent != "" {
```

Comment:
When pulling a remote layer without label(eg, image-ref), the layer degrades to a normal layer and will be completely downloaded. Here the layer is identified as a local block layer and a device will be launched. And the labelKeyTargetSnapshotRef to avoid lauching devices when pulling.